### PR TITLE
[#863] [inference] Add input validation for inference

### PR DESCRIPTION
**Pull request by kparichay**
Add input validation for inference of the neural network

See also #860

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

File: nntrainer/models/neuralnet.cpp [hunk 1]
```diff
@@ -472,9 +499,12 @@ sharedConstTensors NeuralNetwork::inference(sharedConstTensors X) {
     setBatchSize(X[0]->batch());
   }
 
+  sharedConstTensors out;
+  if (!validateInput(X))
+    return out;
```
Comment (zhoonit) [inline nntrainer/models/neuralnet.cpp:504]: how about throw instead of return out?

**Review by zhoonit — APPROVED** [review]
LGTM

File: nntrainer/models/neuralnet.cpp (hunk 1 above)
Comment (kparichay) [reply nntrainer/models/neuralnet.cpp:504]: Line 516 follows the same behavior of return out with a log.
So does train().
I think let's leave it with the same behavior for now and later change all to throw exceptions on error.

Comment (zhoonit) [reply nntrainer/models/neuralnet.cpp:504]: Sounds decent

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Pull request merged**